libr/util/pkcs7.c

```diff
@@ -678,16 +688,22 @@ R_API SpcIndirectDataContent *r_pkcs7_parse_spcinfo(RCMS *cms) {
 	RASN1Object *object = r_asn1_create_object (content->binary, content->length, content->binary);
 	if (!object || object->list.length < 2 || !object->list.objects ||
 		!object->list.objects[0] || !object->list.objects[1]) {
-		r_asn1_free_object (object);
-		free (spcinfo);
-		return NULL;
+		R_FREE (spcinfo);
+		goto beach;
 	}
 	if (object->list.objects[0]) {
-		r_pkcs7_parse_spcdata (&spcinfo->data, object->list.objects[0]);
+		if (!r_pkcs7_parse_spcdata (&spcinfo->data, object->list.objects[0])) {
+			R_FREE (spcinfo);
```

Comment:
Just call free() theres no need to nullify it

Reply:
why not? in beach label spcinfo is returned to the caller. if I will not nullify it then the freed address will be returned to the caller which will assume that the function succeeded.

Reply:
Oh i saw a goto before. Its ok